packages/logs/src/domain/trackNetworkError.ts

```diff
@@ -104,3 +119,67 @@ function format(type: RequestType) {
   }
   return 'Fetch'
 }
+
+/**
+ * Read bytes from a ReadableStream until `limit` bytes have been read.
+ */
+function readBytes(
+  stream: ReadableStream<Uint8Array>,
+  limit: number,
+  callback: (error?: Error, bytes?: Uint8Array) => void
+) {
+  const reader = stream.getReader()
+  const partialBuffers: Uint8Array[] = []
+  let readBytesCount = 0
+
+  readMore()
+
+  function readMore() {
+    reader.read().then(
+      monitor((result: ReadableStreamReadResult<Uint8Array>) => {
```

Comment:
> **🥜 nitpick:**
> 
>     * type can be inferred by ts
In fact, it can't, because of the `monitor` function somehow...
> 
>     * `result` naming does not seem to bring much, we could get rid of it
With destructuring, it can't infer a proper type for the value. By using `result`, it can discriminate the two types
‏
See https://www.typescriptlang.org/play?#code/FAEwpgxgNghgTmABAFwJ4AckCUwGcCuUyiAvIgN6IgD2AdmAFwpz5gA0iAbjFK0-rXAAzAJb0QiAL6IAPhSp1GiIT1zsuPPolzI4YgOZTQkWAmUCIyEXUQBbOiOTU4AHgAqiMAA9kYQbkQACgA6UPh9XCYYWlQAbQBdAEpSAD5EaNQUwKFaJjdEvONoeCQIOh1EdDhqWxE1JgAFatq1FxwCIhTgYAB6HsQAd0cAC2p8YntaR2cOR0QIaMQxITA4dJQMMGAqmrqwYORhv0DAhA7kZJI08mBEO8QzwmRb+77EAD0AfmBJRO63oaHOwOJxwWbEBa0ADkzx2LX2h2Ok2mcBOjyIl2uLzu6Oe90Qby+P0Sf16-VU1A4gOG6yhuKhGl46jmkKWtBWa0RiAABtwmdyNpgGLc4XsDkdaGi8E9MRRsUshEEAIS44I0ejJG74-GqvmseX4t7aon4yTE-79alUSDlXT4SwGcHzaIw7bNMWIyWBcjq9R6sC-VJy-EiRWBJW+zUG+7+6N3QnfU3moA

Reply:
My IDE and gitlab seems to be ok with it though 😕 
cf https://github.com/DataDog/browser-sdk/commit/203c2c85047c3b1663d90583779dbdaa241f5ccf

what am I missing?

Reply:
![Screenshot 2022-01-13 at 16 21 02](https://user-images.githubusercontent.com/61787/149357806-78b7bb62-84d2-47fe-a1cd-304c7fc66e40.png)
There is no error but `value` and `done` are `any`.

Reply:
OK, I have been mislead by my IDE that somehow figure out the types correctly to give type insight for the function parameters 🤔 

<img width="579" alt="Capture d’écran 2022-01-13 à 16 24 24" src="https://user-images.githubusercontent.com/1331991/149358381-da3e59a9-570e-4c77-8f0c-60d6537defa2.png">

Reply:
I've played a bit more with it, it all makes sense now 😅 
Thanks!